pallets/settlement/src/lib.rs

```
@@ -1634,4 +1648,25 @@ impl<T: Config> Module<T> {
         );
         Ok((legs_count, filtered_legs))
     }
+
+    pub fn base_update_venue_signers(
+        did: IdentityId,
+        id: VenueId,
+        signers: Vec<T::AccountId>,
+        update_type: bool,
+    ) -> DispatchResult {
+        // Ensure venue exists & sender is its creator.
+        Self::venue_for_management(id, did)?;
+
+        for signer in &signers {
+            if update_type {
+                <VenueSigners<T>>::insert(&id, &signer, true);
```

If the signer is already a signer it should throw an error.